### PR TITLE
Remove old API code

### PR DESCRIPTION
**Pull request by harrylewis**
# Description

With the introduction of the new library interface, the old code is no longer necessary. This PR removes old code that is no longer actively used.

# Testing

- [x] CI should pass.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.6.6
+2.7.7
```
Comment (harrylewis) [inline .ruby-version:1]: Updating the default Ruby version to align with RuboCop target version.

https://github.com/HiMamaInc/payrix/blob/43d64a2771bb7b04f40e2145e433c7a4b6d29a84/.rubocop.yml#L6-L7

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config`
-# on 2023-08-24 01:23:51 UTC using RuboCop version 1.55.1.
+# on 2023-11-20 14:54:51 UTC using RuboCop version 1.55.1.
```
Comment (harrylewis) [inline .rubocop_todo.yml:3]: I re-ran `rubocop --auto-gen-config` to see if there were any previously ignored rules that are no longer relevant. There were none, but commit this anyways to ensure this remains relevant and current.

File: .rubocop.yml [hunk 3]
```diff
@@ -7,12 +7,6 @@ AllCops:
   TargetRubyVersion: 2.7
   SuggestExtensions: false
   NewCops: enable
-  Exclude:
-    # These files will eventually be removed from the library, so don't bother fixing them up.
-    - 'lib/payrix/http/request_params.rb'
-    - 'lib/payrix/resource/**/*.rb'
-    - 'lib/payrix/resource.rb'
-    - 'spec/lib/payrix/resource/**/*.rb'
```
Comment (harrylewis) [inline .rubocop.yml:15]: All these files have been removed, so no need to ignore them from linting rules.

File: lib/payrix/http.rb [hunk 4]
```diff
@@ -1,9 +1,4 @@
...
 
-module Payrix
-  module Http # rubocop:disable Style/Documentation - Legacy file, which will be removed eventually
-    autoload :RequestParams, 'payrix/http/request_params'
-    autoload :Request, 'payrix/http/request'
-    autoload :Response, 'payrix/http/response'
-  end
-end
+require 'payrix/http/request'
+require 'payrix/http/response'
```
Comment (harrylewis) [inline lib/payrix/http.rb:4]: Move from using `autoload` to `require`, which is more idiomatic of Ruby.

File: lib/payrix/http.rb [hunk 5]
```diff
@@ -1,9 +1,4 @@
 # frozen_string_literal: true
 
-module Payrix
-  module Http # rubocop:disable Style/Documentation - Legacy file, which will be removed eventually
-    autoload :RequestParams, 'payrix/http/request_params'
```
Comment (harrylewis) [inline lib/payrix/http.rb:5]: We have removed `RequestParams`, which is no longer in use.

File: lib/payrix.rb [hunk 6]
```diff
@@ -6,7 +6,6 @@
 require 'payrix/configuration'
 require 'payrix/exceptions'
 require 'payrix/http'
-require 'payrix/resource'
```
Comment (harrylewis) [inline lib/payrix.rb:9]: This is the root for all of the old interface. This file, and all cascading files for `Payrix::Resource::...` have been removed below.

**Review by syedahmedhussaini — APPROVED** [review]
Removals look good to me

**Comment by joncokisler** [comment]
Let's gooo 🎉 🚀

**Pull request merged**